tools/staramr/staramr_search.xml

```diff
@@ -53,32 +57,56 @@
             --plasmidfinder-database-type '${advanced.plasmidfinder_type}'
         #end if
 
-        --output-summary $summary
-        --output-detailed-summary $detailed_summary
-        --output-resfinder $resfinder
-        --output-plasmidfinder $plasmidfinder
-        #if str($use_pointfinder) != 'disabled'
-            --output-pointfinder $pointfinder
-        #end if
-        --output-settings $settings
-        --output-excel $excel_proper_extension
-        --output-mlst $mlst
+        --output-summary '$summary'
+        --output-detailed-summary '$detailed_summary'
+        --output-resfinder '$resfinder'
+        --output-plasmidfinder '$plasmidfinder'
+        --output-settings '$settings'
+        --output-excel '$excel_proper_extension'
+        --output-mlst '$mlst'
 
         --output-hits-dir staramr_hits
 
-        #if str($use_pointfinder) != 'disabled'
-            --pointfinder-organism '${use_pointfinder}'
+        #if str($pointfinder_db.use_pointfinder) != 'disabled'
+            --output-pointfinder '$pointfinder'
+            #if str($pointfinder_db.use_pointfinder) == 'custom'
+                --pointfinder-organism '${pointfinder_db.pointfinder_custom}'
+            #else
+                --pointfinder-organism '${pointfinder_db.use_pointfinder}'
+            #end if
         #end if
 
         $named_genomes
     ]]></command>
     <inputs>
         <param type="data" name="genomes" format="fasta" multiple="true"/>
-        <param name="use_pointfinder" type="select" label="Enable scanning for point mutations using the PointFinder database">
-            <option value="disabled" selected="true">Disable PointFinder</option>
-            <option value="salmonella">Salmonella</option>
-            <option value="campylobacter">Campylobacter</option>
-        </param>
+        <conditional name="pointfinder_db">
+            <param name="use_pointfinder" type="select" label="Enable scanning for point mutations using the PointFinder database">
+                <option value="disabled" selected="true">Disable PointFinder</option>
+                <option value="salmonella">Salmonella</option>
+                <option value="campylobacter">Campylobacter</option>
+                <option value="custom">Unvalidated</option>
+            </param>
+	    <!-- These three values don't have any additional parameters as they are used as-is to control the PointFinder organism -->
+	    <when value="disabled"/>
+	    <when value="salmonella"/>
+	    <when value="campylobacter"/>
```

Comment:
Can you explain this? Where do these end up being used?

Reply:
I only really wish to update the display when a user selects the **Unvalidated** option:

![image](https://user-images.githubusercontent.com/200517/174898657-4e6eb9d3-8be9-4185-a57b-eb72023af072.png)

However, if any other option is selected, then no additional options should be shown:

![image](https://user-images.githubusercontent.com/200517/174899155-ac3b4372-d0a2-4a54-899a-914940dae86b.png)

I do this so that I separate out the validated cases of the PointFinder organism (salmonella, campylobacter) from the unvalidated cases into two separate drop-down menus.

However, the use of a `<conditional>` element in Galaxy (which updates the display of parameters depending on the selected option in the PointFinder organism drop-down) requires a definition of a `<when>` element for every option, even if I don't want to display additional options in these cases. Not including these means that planemo will fail linting of the tool:

```
Applying linter inputs... WARNING
.. WARNING: Conditional [pointfinder_db] no <when /> block found for select option 'disabled'
.. WARNING: Conditional [pointfinder_db] no <when /> block found for select option 'salmonella'
.. WARNING: Conditional [pointfinder_db] no <when /> block found for select option 'campylobacter'
```

So I have to include these three empty `<when>` elements.

Reply:
Aaah. Gotcha. Thank you for the explanation. I didn't know the `when` blocks were mandatory, even when empty.